packages/vite/src/esbuild-request.ts

```diff
@@ -124,6 +151,24 @@ export class EsBuildModuleRequest implements ModuleRequest {
     ) as this;
   }
 
+  private phase: 'bundling' | 'scanning';
+
+  private *extensionSearch(specifier: string): Generator<string> {
```

Comment:
Any integration with `@embroider/core` resolver must do extension search within its defaultResolve. esbuild is not configured with any during the bundling phase, so we have to do it ourselves.